6.bulk_Morphology_Elastic_Network/4.model_performance/scripts/2.model_performance_visualization.r

```diff
@@ -0,0 +1,278 @@
+            "CorrER" = "ER",
+            "CorrGasdermin" = "gasdermin",
+            "CorrPM" = "PM",
+            .default = "other",
+            .missing="other"
+    )
+
+r2_df <- df %>% select(r2)
+r2_df <- unique(r2_df)
+row_ha <- rowAnnotation(
+    df = r2_df,
+    show_legend = TRUE,
+    annotation_name_side = "top",
+    # rotate the title
+    annotation_legend_param = list(
+        title_gp = gpar(fontsize = 12, angle = 0)
+    ),
+    # set color bar for r2 continuous value with brewer palette
+    col = list(r2 = colorRamp2(c(0, 1), c(brewer.pal(9,"BuPu")[1], brewer.pal(9,"BuPu")[7])))
+)
+
+# make the df into a matrix for heatmap
+mat <- dcast(df, feature_names ~ secreted_proteins, value.var = "coefficients")
+row.names(mat) <- mat$feature_names
+mat <- mat %>% select(-feature_names)
+mat <- as.matrix(mat)
+# na to 0
+mat[is.na(mat)] <- 0
+# drop rows that have 0 in 50% of the columns
+mat <- mat[rowSums(mat != 0) > ncol(mat)/2, ]
+mat <- as.data.frame(mat)
+# get the feature names from the index
+mat$feature_names <- row.names(mat)
+# get the feature names for color bar visualization
+features <- mat %>% select(feature_names)
+# drop duplicate features from the feature names
+features <- unique(features)
+features <- features %>%
+        tidyr::separate(
+            feature_names,
+            into = c(
+                "compartment",
+                "feature_group",
+                "measurement",
+                "channel",
+                "parameter1",
+                "parameter2"
+            ),
+            sep = "_",
+            remove = FALSE
+        ) %>%
+        dplyr::mutate(channel_cleaned = channel)
+
+    # Clean channel for visualization
+    features$channel_learned <- dplyr::recode(features$channel,
+            "CorrDNA" = "nuclei",
+            "CorrMito" = "Mito",
+            "CorrER" = "ER",
+            "CorrGasdermin" = "gasdermin",
+            "CorrPM" = "PM",
+            .default = "other",
+            .missing="other"
+    )
+
+# set annotations
+column_ha <- HeatmapAnnotation(
+    Compartment = features$compartment,
+    Feature_Type = features$feature_group,
+    Channel = features$channel_learned,
+    show_legend = TRUE,
+    annotation_legend_param = list(title_position = "topcenter", labels_gp = gpar(fontsize = 12)),
+    annotation_name_side = "right",
+    # color
+    col = list(
+        Compartment = c(
+            "Cells" = brewer.pal(12, "Accent")[5],
+            "Cytoplasm" = brewer.pal(12, "Accent")[6],
+            "Nuclei" = brewer.pal(12, "Accent")[7]
+        ),
+        Feature_Type = c(
+            "AreaShape" = brewer.pal(12, "Paired")[1],
+            "Correlation" = brewer.pal(12, "Paired")[2],
+            "Granularity" = brewer.pal(12, "Paired")[5],
+            "Neighbors" = brewer.pal(12, "Paired")[8],
+            "RadialDistribution" = brewer.pal(12, "Paired")[10],
+            "Texture" = brewer.pal(12, "Paired")[11]
+        ),
+        Channel = c(
```

Comment:
If possible, consider providing an existing color palette to help assign these colors without statically defining them.

Reply:
I manually define the channel colors to reflect the channel wavelengths. I updated the in line documentation to reflect this! :D